reactivex/operators/__init__.py

```diff
@@ -3325,6 +3326,202 @@ def switch_latest() -> Callable[
     return switch_latest_()
 
 
+def switch_map(
+    mapper: Optional[Mapper[_T1, Observable[_T2]]] = None
+) -> Callable[[Observable[_T1]], Observable[_T2]]:
+    """
+    The switch_map operator.
+
+    Project each element of an observable sequence into a new observable.
+
+    .. marble::
+        :alt: switch_map
+
+        ---1---2---3--->
+        [   switch_map(i: of(i, i ** 2, i ** 3))    ]
+        ---1---1---1---2---4---8---3---9---27--->
+
+    Example:
+        >>> switch_map(lambda value: of(value, value // 2))
+
+    Args:
+        mapper: A transform function to apply to each source element.
+
+    Returns:
+        A partially applied operator function that takes an observable
+        source and returns an observable sequence whose elements are
+        each element of the result of invoking the transform function
+        on each element of the source.
+    """
+    from ._switch_map import switch_map_
+
+    return switch_map_(mapper)
```

Comment:
```suggestion
    from ._switchlatest import switch_latest_

    mapper_: typing.Mapper[_T1, Union[Future[_T2], Observable[_T2]]] = mapper or cast(
        Callable[[_T1], Observable[_T2]], of
    )
    return compose(
        map(mapper_),
        switch_latest_(),
    )
```

Reply:
So I don't think we need the full (re)implementation of `switch_map` and we should just reuse `map` and `switch_latest`. This is why is this operator is questionable to add since it's so easy to compose yourself. But I think adding it will help to show that.

Reply:
Interesting, I hadn't really noticed what `switch_latest` was doing. There's so many operators to keep track of.

`mypy` wasn't happy until I made the type value of the cast the same as the annotation of `mapper_`, which makes sense.